### DISCUSSION TITLE
to_image() 编译后报错求助

### DISCUSSION THREAD

**Discussion by jianghwjx** [jsvine/pdfplumber#1051] [Ask for help with specific PDFs]
示例代码：
import pdfplumber

pdf = pdfplumber.open('1.pdf')
for page in pdf.pages:
    page_image = page.to_image()
    page_image.show()

通过python直接运行没有问题，但是使用pyinstaller编译后报错如下
![image](https://github.com/jsvine/pdfplumber/assets/35679970/740dd610-c2ff-4102-8c54-9bc35ef32512)
刚刚接触python，查了很多资料也没明白要如何处理，寻求大家帮助

**Comment by jianghwjx** [comment]
使用的 python3.11
操作系统win11
pdfplumber库使用的 0.10.3

**Comment by wind-chh** [comment]
try to include **pdfium** file when you packaging your script using pyinstaller.

`pyinstaller --onefile --add-binary ".venv/Lib/site-packages/example_package/example.dll;." myscript.py`

https://stackoverflow.com/questions/38791685/how-do-i-include-dll-file-in-executable-using-pyinstaller

**Reply by jianghwjx** [reply]
Thank you very much, but after I tried, I still reported the same error. I found the pdfium.dll I needed at "C:\Users\jiang\AppData\Local\Programs\Python\Python312\Lib\site-packages\pypdfium2_raw\pdfium.dll", using 

> pyinstaller --onefile --add-binary "C:\Users\jiang\AppData\Local\Programs\Python\Python312\Lib\site-packages\pypdfium2_raw\pdfium.dll;." pdf2A5.py

After pdf2A5.py is compiled, it is still the same error.

**Reply by wind-chh** [reply]
I think this is an issue about pyinstaller. I found another post:
https://stackoverflow.com/questions/15114695/imported-module-not-found-in-pyinstaller

and look through the document of pyinstaller [https://pyinstaller.org/en/stable/spec-files.html#adding-data-files](url)

`pyinstaller --add-data "src/README.txt:." myscript.py`

It seems like the "." and ";"/":" matters.
